### DISCUSSION TITLE
Multiple plants

### DISCUSSION THREAD

**Discussion by kamphuisbert** [djansen1987/SAJeSolar#99] [Q&A]
Hi,

Because I can't seem to get the default HA integration to work (https://www.home-assistant.io/integrations/saj/) I tried to setup this solution. It's working fine for 1 plant, but I have no clue how to configure the HACS addon for multiple addons. I tried to just add the same platform as a new sensor, with a different plant_id. In the entities only the instances of the first plant are visible though.

Is there any possibility to set this up so that the sensors of all plants are available?

Regards,
Bert

**Comment by djansen1987** [comment]
Hoi Bert,

Bedankt voor je bericht. Helaas is het met deze integratie nog niet mogelijk om meerdere plants tegelijkertijd uit te lezen. Mocht je handig zijn met programmeren, dan is een aanpassing altijd welkom op de bestaande code. Dit kun je doen via een pull request. Zelf heb ik helaas geen toegang meer tot SAJ, dus ik kan het helaas niet testen of verder ontwikkelen.

Je zou ook kunnen kijken naar de integratie van @faanskit. Deze ondersteunt dat wel, maar helaas niet alle modellen. Ook staat de ontwikkeling bij faanskit stil omdat hij ook geen saj zelf meer heeft.

**English Translation:**

Hi Bert,

Thank you for your message. Unfortunately, it is not yet possible with this integration to read multiple plants simultaneously. If you are handy with programming, an adjustment is always welcome on the existing code. You can do this via a pull request. Unfortunately, I no longer have access to SAJ myself, so I can't test or develop it further.

You might also want to look into the integration by @faanskit. This one does support multiple plants but does not support all models. 

Is there anything else you need help with?

**Reply by djansen1987** [reply]
https://github.com/faanskit/ha-esolar
